Счётчик, Сломанная карта, Калькулятор, Отмеченные Email-ы, Выбранный митап

DaliyaAsel opened the pull request:
Счётчик
Сломанная карта
Калькулятор
Отмеченные Email-ы
Выбранный митап

DaliyaAsel committed:
Счётчик
Сломанная карта
Калькулятор
Отмеченные Email-ы
Выбранный митап

jsru-1 commented:
Добавляю преподавателя (@ShGKme) для код-ревью.

ShGKme commented on 01-basics/20-weather/WeatherApp.js:
```diff
@@ -4,44 +4,81 @@ import { getWeatherData, WeatherConditionIcons } from './weather.service.ts'
+      return  WeatherConditionIcons[String(iconId)];
+    }
+
+    const isSunset = (sunset) => {
+      const [hour] = sunset.split(':');
+      return Math.round(hour) >= 19 || Math.round(hour) <= 5;
+    }
```
Не критично, но здесь используются "случайные" числа 19 и 5 вместо времени восхода и заката в текущем месте, при этом минуты игнорируются.

DaliyaAsel replied:
Поняла, спасибо

ShGKme commented on 02-basics-2/10-counter/CounterApp.js:
```diff
@@ -1,25 +1,58 @@
 
-  setup() {},
+  setup() {
+    const counter = ref({
+      count: 0
+    })
```
Оборачивание в объект со свойство `count` здесь лишнее. Достаточно просто хранить число.

DaliyaAsel replied:
Исправлено

ShGKme commented on 02-basics-2/20-broken-map/MapApp.js:
```diff
@@ -1,39 +1,38 @@
+      x.value = event.offsetX
+      y.value = event.offsetY
+      // Обновляем стиль метки после изменения координат
+      if (pinRef.value) {
+        pinRef.value.style.left = `${x.value}px`
+        pinRef.value.style.top = `${y.value}px`
+      }
```
Одна проблема в этой задаче справлена корректно, но вторая осталась - это прямая мутация DOM, за который отвечает Vue.

Согласно шаблону, здесь всегда просто рендерится `span`. Но фактически при изменении состояния приложения, меняется состояние этого `span`-а, меняются его стили.

Они должны быть также описаны в шаблоне через привязку. Тогда шаблон будет сразу декларировать, что положение этого пина определяется таким-то состоянием.

DaliyaAsel replied:
Исправлено

ShGKme commented on 02-basics-2/30-calculator/CalculatorApp.js:
```diff
@@ -1,26 +1,55 @@
 
-  setup() {},
+  setup() {
+    const inputNumber = ref({
+      first: 0,
+      second: 0,
+    })
```
Здесь также лучше описывать 2 отдельные переменные, чем общий объект, если с ним не нужно работать, как с единым целым.

DaliyaAsel replied:
Исправлено

ShGKme commented on 02-basics-2/50-selected-meetup/SelectedMeetupApp.js:
```diff
@@ -1,75 +1,88 @@
-          <div class="radio-group__button">
+          <div
+            class="radio-group__button"
+            v-for="id in [1,2,3,4,5]"
```
`v-for="id in [1,2,3,4,5]"` тоже что `v-for="id in 5"`

DaliyaAsel replied:
Исправлено

ShGKme reviewed (changes requested):
Требуется исправление в `02-basics-2/20-broken-map`, остальные задачи приняты

DaliyaAsel committed:
Счётчик
Сломанная карта
Калькулятор
Отмеченные Email-ы
Выбранный митап. Правки после ревью

jsru-1 commented:
Добавляю преподавателя (@ShGKme) для код-ревью.

DaliyaAsel committed:
Счётчик
Сломанная карта
Калькулятор
Отмеченные Email-ы
Выбранный митап. Правки после ревью

DaliyaAsel committed:
Merge remote-tracking branch 'origin/master'

jsru-1 commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
MeetupView
UiClock
Удаляемые Email-ы
Компонент-счётчик
Компоненты: погода в Средиземье

jsru-1 commented:
Решение было обновлено, посмотрим что скажет @ShGKme

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
Merge branch 'master' of https://github.com/js-tasks-ru/vue-20240924_theblackdahlia

DaliyaAsel committed:
MeetupView в `script setup`
MeetupCover с v-bind в CSS
UiStretch

jsru-1 commented:
Решение было обновлено, посмотрим что скажет @ShGKme

ShGKme commented on 04-sfc/20-MeetupCover-style-v-bind/MeetupCover.vue:
```diff
@@ -24,9 +23,7 @@ const bgStyle = computed(() => (props.image ? { '--bg-url': `url('${props.image}
-  /* Если изображение присутствует - берём его из CSS переменной, установленной на элемент в шаблоне */
-  /* Иначе выводим изображение по умолчанию - var(--default-cover) */
-  background-image: linear-gradient(0deg, rgba(0, 0, 0, 0.4), rgba(0, 0, 0, 0.4)), var(--bg-url, var(--default-cover));
+  background-image: v-bind('backgroundImage ? props.image : 'var(--default-cover)'');
```
Свойства `backgroundImage` в компоненте нет.
И потерялся первый фон с линейным градиентом затемнения.

В целом лучше такие вычисления делать в `setup`, чтобы не перегружать стили вычислением и не генерировать после сборки CSS переменных со страшным именем.

ShGKme commented on 03-components/30-removable-emails/EmailList.js:
```diff
@@ -15,13 +15,16 @@ export default defineComponent({
         :key="email"
         :email="email"
         :marked="isMarked"
+        @remove="() => $emit('removeEmailByIndex', index)"
```
У директивы `v-on` можно передавать сразу тело обработчика события
`@remove="$emit('removeEmailByIndex', index)"`

ShGKme commented on 03-components/30-removable-emails/MarkedEmailsApp.js:
```diff
@@ -69,7 +69,7 @@ export default defineComponent({
         <UiInput v-model.trim="query" type="search" placeholder="Поиск" aria-label="Поиск" small />
       </UiFormGroup>
-      <EmailList :emails="markedEmails" />
+      <EmailList :emails="markedEmails" @removeEmailByIndex="removeEmailByIndex"/>
```
В шаблоне принято события и пропсы использовать в `kebab-case`

ShGKme commented on 03-components/50-weather-components/WeatherListItem.js:
```diff
@@ -0,0 +1,89 @@
+      return  WeatherConditionIcons[String(iconId)];
+    }
+
+    const isSunset = (sunset) => {
+      const [hour] = sunset.split(':');
+      return Math.round(hour) >= 19 || Math.round(hour) <= 5;
+    }
```
Видимо, не обратил на это внимание в прошлой версии задачи. Не критично, но здесь:
1. Не учитывается время в текущем населённом пункте. Проверяется только время захода, при чём что время захода солнца после 19 или до 5 утра. 
2. Не считываются минуты

ShGKme commented on 04-sfc/30-UiStretch/UiStretch.vue:
```diff
@@ -7,14 +8,19 @@
 </template>
 
 <style scoped>
+.stretch-container {
+  width: 100%;
+  height: 100%;
+}
```
Это новое правило меняет работу компонента. Размер определяет родитель по своему усмотрению, а этот компонент растягивает только содержимое.

ShGKme commented on 04-sfc/30-UiStretch/UiStretch.vue:
```diff
@@ -7,14 +8,19 @@
-.stretch-container > img,
-.stretch-container > video,
-.stretch-container > picture {
+.stretch-container :deep img,
+.stretch-container :deep video,
+.stretch-container :deep picture {
```
Ранее правило здесь действовало только на непосредственно дочернюю картинку, переданную в этот компонент. Теперь `>` потерялся, и действует глубоко на сколько угодно глубоко лежащую картинку

ShGKme replied:
Также корректный синтаксис - как с псевдо селекторами через скобки : `:deep(img)`
